### PR TITLE
Fix for BLAS dot library node

### PR DESCRIPTION
**Pull request by alexnick83**
The return value of `cblas_?dot` is supposed to be a scalar and not a pointer.
Fix by changing the OpenBLAS expansion tasklet code from `_result = cblas ...` to `_result[0] = cblas ...`.
`_result` is an array of size 1.

### CODE REVIEW

**Review by tim0s — APPROVED** [review]
this should work - let's see if the tests pass :)

**Comment by tbennun** [comment]
@alexnick83 shouldn't you change the connector type in the tasklet to scalar instead? This might fail in the cases where the output is a scalar

**Review by tbennun — CHANGES_REQUESTED** [review]
see message

File: tests/library/blas_dot_test.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def make_sdfg(implementation, dtype, storage=dace.StorageType.Default):
     state.add_memlet_path(dot_node,
                           result,
                           src_conn="_result",
-                          memlet=Memlet.simple(result, "0", num_accesses=-1))
+                          memlet=Memlet.simple(result, "0", num_accesses=1))
```
Comment (alexnick83) [inline tests/library/blas_dot_test.py:55]: Major issue: If this goes back to -1 (memlet dynamic), code generation does not generate code (see dace/codegene/targets/cpu.py, line 837)

Comment (tbennun) [reply tests/library/blas_dot_test.py:55]: it shouldn't be dynamic anyway - this is by design. The code generator does not output anything on its own if `dynamic=True`. Instead, this is done in the tasklet.

**Review by tbennun — APPROVED** [review]

**Pull request merged**